portalnet/src/overlay/service.rs

```diff
@@ -1574,19 +1584,26 @@ where
             .await
             .into_iter()
             .enumerate()
-            .filter_map(|(index, content)| content.unwrap_or_else(|err| {
-                // Extract the error from the thread handle
-                let err = err.into_panic();
-                let err = if let Some(err) = err.downcast_ref::<&'static str>() {
-                    err.to_string()
-                } else if let Some(err) = err.downcast_ref::<String>() {
-                    err.clone()
-                } else {
-                    format!("{err:?}")
-                };
-                debug!(err, content_key = ?content_keys_string[index], "Process uTP payload tokio task failed:");
-                None
-            }))
+            .filter_map(|(index, content)| match content {
+                Ok(content) => {
+                    offer_queue.write().successful_processing(&content_keys[index]);
+                    content
+                }
+                Err(err) => {
+                    offer_queue.write().failed_to_process(&content_keys[index]);
```

Comment:
Reading this doesn't imply that we are sending overlay requests inside `failed_to_process` which is a hidden side effect of this function. Why do we not send the `FIndContent` request from here?

Reply:
Well, to be honest, it was primarily to reduce the size of `OverlayService` but I see your point and agree that it makes more sense to handle this inside the service, regardless of the additional code

Reply:
Though, handling this inside the `OverlayService` does make it significantly more difficult to test... I do still agree that it makes more sense